go/service/teams.go

```diff
@@ -60,6 +62,83 @@ func (h *TeamsHandler) TeamCreate(ctx context.Context, arg keybase1.TeamCreateAr
 	return h.TeamCreateWithSettings(ctx, arg2)
 }
 
+func (h *TeamsHandler) TeamCreateFancy(ctx context.Context, arg keybase1.TeamCreateFancyArg) (teamID keybase1.TeamID,
+	err error) {
+	ctx = libkb.WithLogTag(ctx, "TM")
+	mctx := libkb.NewMetaContext(ctx, h.G().ExternalG())
+	teamInfo := arg.TeamInfo
+	defer h.G().CTraceTimed(ctx, fmt.Sprintf("TeamCreateFancy(%s)", teamInfo.Name), func() error { return err })()
+
+	arg2 := keybase1.TeamCreateWithSettingsArg{
+		Name:        teamInfo.Name,
+		SessionID:   arg.SessionID,
+		Settings:    teamInfo.OpenSettings,
+		JoinSubteam: teamInfo.JoinSubteam,
+	}
+	teamCreateRes, err := h.TeamCreateWithSettings(ctx, arg2)
+	teamID = teamCreateRes.TeamID
+	if err != nil {
+		return teamID, err
+	}
+	var errs []error
+
+	err = teams.SetTeamShowcase(ctx, h.G().ExternalG(), teamID, &teamInfo.Showcase, &teamInfo.Description, nil)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if teamInfo.Avatar != nil {
+		avatar := teamInfo.Avatar
+		arg3 := keybase1.UploadTeamAvatarArg{Teamname: teamInfo.Name, Filename: avatar.AvatarFilename,
+			Crop: avatar.Crop, SendChatNotification: false}
+		err = teams.ChangeTeamAvatar(mctx, arg3)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	for _, topicName := range teamInfo.ChatChannels {
+		_, _, err = h.G().ChatHelper.NewConversation(ctx, gregor1.UID(h.G().GetMyUID().ToBytes()), teamInfo.Name,
+			&topicName, chat1.TopicType_CHAT, chat1.ConversationMembersType_TEAM, keybase1.TLFVisibility_PRIVATE)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	for _, subteamName := range teamInfo.Subteams {
+		name := teamInfo.Name + "." + subteamName
+		_, err = h.TeamCreate(ctx, keybase1.TeamCreateArg{SessionID: arg.SessionID, Name: name})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(teamInfo.Users) > 0 {
+		arg4 := keybase1.TeamAddMembersMultiRoleArg{
+			SessionID:            arg.SessionID,
+			TeamID:               teamID,
+			Users:                teamInfo.Users,
+			SendChatNotification: false,
+			EmailInviteMessage:   teamInfo.EmailInviteMessage,
+		}
+
+		unaddedUsers, err := h.TeamAddMembersMultiRole(ctx, arg4)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if unaddedUsers.NotAdded != nil {
+			errs = append(errs, fmt.Errorf("could not add members to team: %v", unaddedUsers.NotAdded))
```

Comment:
I think you need a length check. This keeps happening with an empty array for me.